### ISSUE TITLE
Searching with same keywords (different order) display distinct results

### ISSUE THREAD

**Issue by diogo405** [BraydenTW/react-emoji-search#8]
### What is this About?

Search is presenting different results for same query keywords

<br />

### Expected Behavior

It should show same results.

<br />

### Steps to Reproduce

1. type in: smile cat
2. it shows the smirk cat emoji
3. clear the input and type in: cat smile
4. it doesn't show any results

<br />

![Screen Shot 2020-10-07 at 10 16 48 am](https://user-images.githubusercontent.com/691570/95273653-f4c31e80-0886-11eb-8139-900ebed33bd2.png)

<br />

![Screen Shot 2020-10-07 at 10 16 36 am](https://user-images.githubusercontent.com/691570/95273689-05739480-0887-11eb-88e3-66f5d436c3ee.png)

**Comment by BraydenTW** [comment]
Thanks again for your contribution @diogo405. All very helpful and vital information to help this OSS project out 👍 

I would be happy to take this on unless you believe you have a much quicker solution.

**Comment by diogo405** [comment]
> Thanks again for your contribution @diogo405. All very helpful and vital information to help this OSS project out 👍
> 
> I would be happy to take this on unless you believe you have a much quicker solution.

Thanks! Happy to help 😀
Yep, go for it mate.

**Comment by BraydenTW** [comment]
Sounds great, thanks!

**Comment by BraydenTW** [comment]
@diogo405, I'm having a bit of trouble with the searching algorithim. Right now I am using a simple filter that looks like this:

```js
<Results emojiFiltered={search === "" ? emojiData : emojiData.filter(item => item.keywords.includes(search))} />
```

It takes the searched value and filters out any item that doesn't have a keyword with that value.

However that was giving problems like you mentioned earlier:

> ### Steps to Reproduce
> 1. type in: smile cat
> 2. it shows the smirk cat emoji
> 3. clear the input and type in: cat smile
> 4. it doesn't show any results
> ![Screen Shot 2020-10-07 at 10 16 36 am](https://user-images.githubusercontent.com/691570/95273689-05739480-0887-11eb-88e3-66f5d436c3ee.png)

Do you have some ideas of how to better filter the results to get more accurate results?

**Comment by diogo405** [comment]
Yeah, that's a bit tricky hehe. I think if you modify the emoji item keyword to be an array and split the search query might work. Something like this:

```
// Some tests here
//const query = 'cat smile'
//const query = 'smile cat '
//const query = 'smile cat h'
//const query = 'smile cat ha'
//const query = 'h   '
//const query = 'ha'
const query = 'hat'

// Didn't include all props to make it smaller
const emojiData = [
  {"keywords": ["cat", "smile", "heart"]},
  {"keywords": ["cat", "smile", "happy"]}
]

// Split the query and loop it
const queryKeywords = query.trim().split(' ')
let emojisFound = new Set()
queryKeywords.forEach(queryKeyword => {

	// For each query keyword check if emojiData item keywords match it - that's the hardest to understand, try to read from inside to outside maybe?!
	let filtered = emojiData.filter(item => item.keywords.some(emojiKeyword => emojiKeyword.indexOf(queryKeyword) !== -1))

	// Add to set
	filtered.forEach(item => emojisFound.add(item))
})

console.log([...emojisFound])
```

**Comment by BraydenTW** [comment]
Much better idea than the one I had. I’ll work on implanting these changes soon.

Thanks again! 🙏

**Comment by BraydenTW** [comment]
@diogo405, I am working on implementing that new method you just introduced to me. However, I don't understand how you think I should break up the search query. I tried doing it this way:

If search query is `smile`, add these to the query array:

```js
[
"s",
"sm",
"smi",
"smil",
"smile"
]
```

Howevery a query like this would bring things like `shamrock` in the results because it matches with `"s"`.

Should we just stick with keeping the search value to its original?

**Comment by diogo405** [comment]
I think you need to break the search by keywords. If the query is "smile" so your keywords array would be ["smile"], just one keyword. On the other hand, if it was "cat smile" the keywords array would be ["cat", "smile"]. In this way you can check every query keyword against each emoji list of keywords.

You can get the keywords array using the split method:
```
const query = "cat smile happy"
const keywords = query.split(' ') // there's a space between the quotes here
console.log(keywords) // ["cat", "smile", "happy"]
```

**Comment by BraydenTW** [comment]
Hey @diogo405 , I just updated the search with some of your tips from earlier and submitted + merged the PR (#10).

Let me know what you think.

**Comment by diogo405** [comment]
Nice you fixed 👍🏽 Another thing we could do is to separate the app css so each component has its css file (instead of having App.css with all classes). What do you think?
